### PR TITLE
Fix scope details in the invocation context.

### PR DESCRIPTION
**Pull request by ShalkiWenushika**
### Purpose
This PR will add correct scope details to the invocation context. 

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/1498

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Comment by VirajSalaka** [comment]
Shall we create a git issue and link it with the PR?

File: components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/oauth2_key_validation_provider.bal [hunk 1]
```diff
@@ -219,6 +219,24 @@ public type OAuth2KeyValidationProvider object {
 
     }
 
+    public function setScopesToPrincipalComponent(xml responseXml) {
+            string[] scopes = [];
+            int i = 0;
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/oauth2_key_validation_provider.bal:224]: Lets use proper variables names

File: components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/oauth2_key_validation_provider.bal [hunk 2]
```diff
@@ -219,6 +219,24 @@ public type OAuth2KeyValidationProvider object {
...
+            // Iterate through scopes.
+            foreach xml|string scope in responseXml/<apim:scopes>/* {
+                if (scope is xml) {
```
Comment (Rajith90) [inline components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/oauth2_key_validation_provider.bal:227]: If scope is string then what are we returning

Comment (ShalkiWenushika) [reply components/micro-gateway-core/src/main/ballerina/src/gateway/handler_providers/oauth2_key_validation_provider.bal:227]: Added an error message if scope is string

**Comment by ShalkiWenushika** [comment]
> Shall we create a git issue and link it with the PR?

**Comment by bhathiya** [comment]
@ShalkiWenushika why is this still open?

**Review by Rajith90 — APPROVED** [review]

**Pull request merged**